app/Http/Controllers/api/v1/RoomController.php

```diff
@@ -120,65 +120,59 @@ public function start(Room $room, StartJoinMeeting $request)
         $meeting = $room->runningMeeting();
         if (!$meeting) {
             // Create new meeting
-            $meeting              = $room->meetings()->create();
+            $meeting              = new Meeting();
             $meeting->start       = date('Y-m-d H:i:s');
             $meeting->attendeePW  = bin2hex(random_bytes(5));
             $meeting->moderatorPW = bin2hex(random_bytes(5));
+
+            // Basic load balancing, get server with lowest usage
+            $server = Server::lowestUsage();
+
+            // If no server found, throw error
+            if ($server == null) {
+                abort(CustomStatusCodes::NO_SERVER_AVAILABLE, __('app.errors.no_server_available'));
+            }
+
+            $meeting->server()->associate($server);
+            $meeting->room()->associate($room);
             $meeting->save();
```

Comment:
Why not save the meeting only if everything is successful?

Reply:
In https://github.com/THM-Health/PILOS/blob/64-live-data-statistics/app/Meeting.php#L85 the id of the meeting is needed.